lib/pbench/test/unit/server/test_server_configuration.py

```diff
@@ -0,0 +1,220 @@
+from http import HTTPStatus
+
+import pytest
+import requests
+
+
+class TestServerConfiguration:
+    @pytest.fixture()
+    def query_get(self, client, server_config):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+        """
+
+        def query_api(
+            key: str, expected_status: HTTPStatus = HTTPStatus.OK
+        ) -> requests.Response:
+            k = f"/{key}" if key else ""
+            response = client.get(f"{server_config.rest_uri}/server/configuration{k}")
+            assert response.status_code == expected_status
+            return response
+
+        return query_api
+
+    @pytest.fixture()
+    def query_put(self, client, server_config, pbench_admin_token):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+            pbench_admin_token: Provide an admin authentication token
+        """
+
+        def query_api(
+            key: str,
+            expected_status: HTTPStatus = HTTPStatus.OK,
+            token: str = pbench_admin_token,
+            **kwargs,
+        ) -> requests.Response:
+            k = f"/{key}" if key else ""
+            response = client.put(
+                f"{server_config.rest_uri}/server/configuration{k}",
+                headers={"authorization": f"bearer {token}"},
+                **kwargs,
+            )
+            assert response.status_code == expected_status
+            return response
+
+        return query_api
+
+    def test_get_bad_keys(self, query_get):
+        response = query_get("xyzzy", HTTPStatus.BAD_REQUEST)
+        assert response.json == {
+            "message": "Unrecognized keyword ['xyzzy'] given for parameter key; allowed keywords are ['dataset-lifetime', 'server-banner', 'server-state']"
+        }
+
+    def test_get1(self, query_get):
+        response = query_get("dataset-lifetime")
+        assert response.json == {"dataset-lifetime": "3650"}
+
+    def test_get_all(self, query_get):
+        response = query_get(None, HTTPStatus.OK)
```

Comment:
To correct this time traveling mis-impression... at some point along the way I *added* the default value to `query_[get|put]` so I didn't have to specify the normal value everywhere, but I didn't bother to go back and remove it from every call where it would now have been possible.

Reply:
Wibbly wobbly.

I, of course, tripped over this one because it had a green background.  But, in general, there's a three-state logic, where the caller accepts the default, specifies some other value, or explicitly specifies the default, and there are good reasons for each of these approaches.  However, when the caller takes the last option, it's important to ensure that that option is the correct one.  Taking the first option means that the caller is delegating the choice of value to the callee; whereas the last option ensures that the callee is _not_ the one to make the choice.  And, in the case of the last option, we need to be sure that we'll continue to get the correct result even if we change the default.  Otherwise, the caller has taken the wrong option!

Now, admittedly, in this particular case, there is nothing wrong with taking the third option.  But, in general, I look askance at code which explicitly specifies what is the default option...because that means that it will not benefit from a future change of the default without us also finding and "correcting" that call...and so it looks like a latent bug, to me.